march_joint_inertia_controller/include/march_joint_inertia_controller/joint_trajectory_inertia_controller.h

```diff
@@ -0,0 +1,209 @@
+      if (!pids_[i]->init(joint_nh))
+      {
+        ROS_WARN_STREAM("Failed to initialize PID gains from ROS parameter server.");
+        return false;
+      }
+    }
+
+    // Obtain inertia estimator parameters from server
+    double lambda[2];
+    int alpha_filter_size[2];
+    ros::NodeHandle rotary_estimator_nh(nh, std::string("inertia_estimator/rotary"));
+    rotary_estimator_nh.param("std_samples", samples_, 100);
+    rotary_estimator_nh.param("lambda", lambda[0], 0.96);
+    rotary_estimator_nh.param("alpha_filter_size", alpha_filter_size[0], 12);
+
+    ros::NodeHandle linear_estimator_nh(nh, std::string("inertia_estimator/linear"));
+    linear_estimator_nh.param("lambda", lambda[1], 0.96);
+    linear_estimator_nh.param("alpha_filter_size", alpha_filter_size[1], 12);
+
+    // Initialize the estimator parameters
+    inertia_estimators_.resize(num_joints_);
+    for (unsigned int j = 0; j < num_joints_; ++j)
+    {
+      inertia_estimators_[j].setNodeHandle(nh);
+      inertia_estimators_[j].setLambda(lambda[(int)floor(j / 2) % 2]);  // Produce sequence 00110011
+      inertia_estimators_[j].setAcc_size(alpha_filter_size[(int)floor(j / 2) % 2]);
+      inertia_estimators_[j].configurePublisher(joint_handles[j].getName());
+    }
+    return true;
+  }
+
+  /**
+   * \brief Updates the commanded effort for each individual joint and estimates the inertia on each joint and publishes
+   * that information on a topic
+   */
+  void updateCommand(const ros::Time& /*time*/, const ros::Duration& period,
+                     const joint_trajectory_controller::State& /*desired state*/,
+                     const joint_trajectory_controller::State& state_error)
+  {
+    num_joints_ = joint_handles_ptr_->size();
+
+    // Preconditions
+    if (!joint_handles_ptr_)
+    {
+      return;
+    }
+    assert(num_joints_ == state_error.position.size());
+    assert(num_joints_ == state_error.velocity.size());
+
+    if (count_ < samples_)
+    {
+      count_++;
+      for (unsigned int i = 0; i < num_joints_; ++i)
+      {
+        inertia_estimators_[i].fill_buffers((*joint_handles_ptr_)[i].getVelocity(),
+                                            (*joint_handles_ptr_)[i].getEffort(), period);
+        inertia_estimators_[i].standard_deviation.push_back(inertia_estimators_[i].getAcceleration(0));
+        if (count_ == samples_)
+        {
+          inertia_estimators_[i].init_p(samples_);
+        }
+      }
+    }
+    else
+    {
+      for (unsigned int i = 0; i < num_joints_; ++i)
+      {
+        // Update inertia estimator
+        inertia_estimators_[i].fill_buffers((*joint_handles_ptr_)[i].getVelocity(),
+                                            (*joint_handles_ptr_)[i].getEffort(), period);
+        inertia_estimators_[i].inertia_estimate();
+        inertia_estimators_[i].publishInertia();
+        // TO DO: Provide lookup table for gain selection
+        // TO DO: apply PID control
```

Comment:
I couldn't find Jira issues for this, is that correct? If so I think you should add them, so they are not forgotten.

Reply:
That was correct, I have added them recently